### ISSUE TITLE
[docs] Switch NJK to WEBC for layout

### ISSUE THREAD

**Issue by markcaron** [RedHat-UX/red-hat-design-system#1428]
Do it.

![image](https://github.com/RedHat-UX/red-hat-design-system/assets/1890950/32542241-d16f-4e7a-afbd-47f4553f538b)

**Comment by bennypowers** [comment]
i'm concerned about webc's lack of a satisfying dsd support. as an alternative to webc, I would like us to consider lit-11ty

**Comment by zeroedin** [comment]
> lit-11ty

this? https://github.com/lit/lit/tree/main/packages/labs/eleventy-plugin-lit

**Comment by bennypowers** [comment]
yes, that or something like it. the idea being that instead of webc components we'd write uxdot-specific lit components and ssr them by way of dsd, using the plugin

**Comment by zeroedin** [comment]
Do we need a list of `uxdot/elements` that we are going to create for this?  Elements that aren't quite RHDS elements but need a home.

Here is a shortlist that exist or might be additionally needed (with screenshots of specific parts I'm referencing):

- masthead ![image](https://github.com/RedHat-UX/red-hat-design-system/assets/13666/85f07877-7f4d-432e-b9a8-034183ab00ac) will need to [include github link](https://github.com/RedHat-UX/red-hat-design-system/issues/1440) 
- article/page header 
![image](https://github.com/RedHat-UX/red-hat-design-system/assets/13666/8d454a8a-d20a-4120-bc51-fef7773aff34)
- sample/example - already being worked on in #1350 
- installation-tabs - already live 

Anyone have any other elements on the page they think might serve well as components?

**Comment by zeroedin** [comment]
Given inclusion with 11ty-plugin-lit,  how do we handle eleventy based templating? Are we still using njk for that?  Do we just pair that down as much as possible what can go into a component and what can't?

Example: I want to loop through 11ty's data cascade?

**Comment by bennypowers** [comment]
we already have docs/assets/elements (iirc) and more in the docs: sample pr

**Comment by bennypowers** [comment]
https://github.com/lit/lit/tree/main/packages/labs/eleventy-plugin-lit#passing-data-to-components

**Comment by zeroedin** [comment]
> https://github.com/lit/lit/tree/main/packages/labs/eleventy-plugin-lit#passing-data-to-components

Ok so the idea is there but no actual way to do that yet.  

If we are to get this done this quarter, then that doesn't look promising given the timeline. 

 So will we have to fall back to something like:  `<some-component name="{{ data.name}}">` that is still powered by njk?  After compile then passing the compiled html output to the plugin for DSD'ing?

**Comment by bennypowers** [comment]
maybe https://github.com/lit/lit/issues/2485#issuecomment-1865197284

**Comment by markcaron** [comment]
Going a different route with components.

**Comment by zeroedin** [comment]
Cleanup for potential move to using elventy-lit-plugin has been done in the docs rewrite PR: https://github.com/RedHat-UX/red-hat-design-system/pull/1465
